llvm/include/llvm/IR/VectorBuilder.h

```diff
@@ -15,6 +15,7 @@
 #ifndef LLVM_IR_VECTORBUILDER_H
 #define LLVM_IR_VECTORBUILDER_H
 
+#include <llvm/Analysis/IVDescriptors.h>
```

Comment:
This is a layering violation.

Reply:
Thanks for pointing out this issue. I opened #99276 to fix it.  Please take a look, thanks a lot.